src/Dnsimple/Client.php

```diff
@@ -73,16 +152,57 @@ public function delete($path, array $options = [])
         return $this->request("DELETE", $path, $options);
     }
 
+    public function patch($path, array $options = [])
+    {
+        return $this->request("PATCH", $path, $options);
+    }
+
+    public function put($path, array $options = [])
+    {
+        return $this->request("PUT", $path, $options);
+    }
+
     public function request($method, $path, array $options = [])
     {
         $requestOptions = array_merge_recursive($options, [
             "headers" => [
                 "Authorization" => "Bearer $this->accessToken",
                 "Accept" => "application/json",
-                "User-Agent" => "dnsimple-php/".VERSION,
+                "User-Agent" => $this->getUserAgent(),
             ]
         ]);
 
-        return $this->httpClient->request($method, $path, $requestOptions);
+        try {
+            return $this->httpClient->request($method, $path, $requestOptions);
+        } catch (GuzzleException $e) {
+            throw new DnsimpleException($e);
+        }
+    }
+
+    public function getUserAgent(): string
+    {
+        return trim($this->customUserAgent . " " . self::DEFAULT_USER_AGENT);
+    }
+
+    public function setUserAgent($customName)
```

Comment:
Can you please make sure public methods have docblock?

Reply:
I think in a few cases I totally forgot to add comments given the sheer amount of comments needed in the project 😅